lib/jwt/decode.rb

```diff
@@ -61,9 +62,12 @@ def verify_algo
     def set_key
       @key = find_key(&@keyfinder) if @keyfinder
       @key = ::JWT::JWK::KeyFinder.new(jwks: @options[:jwks], allow_nil_kid: @options[:allow_nil_kid]).key_for(token.header['kid']) if @options[:jwks]
-      return unless (x5c_options = @options[:x5c])
 
-      @key = X5cKeyFinder.new(x5c_options[:root_certificates], x5c_options[:crls]).from(token.header['x5c'])
+      if (x5c_options = @options[:x5c])
```

Comment:
If we have a namespace for the keyfinders the resolving could happen in that.

`@key = KeyFinders.resolve_key(@options, token)`